### DISCUSSION TITLE
Need some help with Product Class structure

### DISCUSSION THREAD

**Discussion by lukutisz** [pimcore/pimcore#12614] [Q&A / Support]
Hello,

I am having some issues in understanding how to achieve a wanted result of Product Class structure.

We have products, all of these products are having shared attributes such as Name, SKU, Description, Images, but as all of these products are different type of products, they have some product type specific attributes.

What would be the best approach for having product specific attributes like Attribute sets, if Product type is Shirts, then in this case it is required to enter Shirts related attributes, if its Pants, then it would show Pants related attributes and etc. 

Of course Object Bricks could be used for that, but it would also allow to add all of these Product types at the same time.

We don't want to use separate Classes for each Product Type as it would also make no sense.

And also later in Export we will have to have a field that identifies what type of product is that, as example Shirts and it will have to contain all Shirt related attributes.

Maybe there is somekind of easy approach for that? Hope the idea is clear.

Some example images that are the closest outcome:
![image](https://user-images.githubusercontent.com/66670706/177875443-3e5786be-5147-472b-a4bb-259abbc65d09.png)
![image](https://user-images.githubusercontent.com/66670706/177875506-52fe2535-2d4b-4fe8-adf0-615a75d1234d.png)

**Comment by MadeP** [comment]
You could use ObjectBricks and set max items to 1 under specific settings.  This way only one type can be assigned to a product. The type of the product can be set via the ObjectBrick with an invisible text field that contains the corresponding type by default.

**Reply by passioneight** [reply]
I'd also go for this approach, except that I wouldn't use a field to select a type, but have the brick class reflect it. That is, I'd create the `Shirt` and `Pants` brick, add a class override for the `Product`, and implement a `isShirt` and `isPants` method.
